### ISSUE TITLE
Include latest changes in this repository

### ISSUE THREAD

**Issue by gqzomer** [HorstBaerbel/MixxxReloopMixageMappings#1]
I was looking for a mixxx mapping for my reloop mixage and found this repository. I thought you laid a pretty good foundation but found that some features like the search/scratch mode of the jog wheels were not working quite right. To address this and make some further improvements I made a fork in the hopes of contributing to the mapping you created. Only to find out later that you continued working to improve the mapping in a fork of the mixxx repository found [here](https://github.com/HorstBaerbel/mixxx/tree/reloop_mixage). Maybe you can update this repo to include the later changes you made or link to the repo you forked in the readme. Currently this is repo is one of the first results when you google 'mixxx reloop mixage mappings'. 

Im currently still working on expanding the effect controls and support for controlling deck 3 and 4 with the reloop mixage. Should I make a pull request on this repo or instead make pull request on your reloop_mixage fork?

**Comment by HorstBaerbel** [comment]
Hi gqzomer. I'm not sure what you're missing or if you've even got the latest version. The [pull request for Mixxx](https://github.com/mixxxdj/mixxx/pull/10892) is still pending and afair the reloop_mixage repo is VERY different to this one (I should probably archive this one). See the pull request for docs too. I tried to map the controller as faithful to the its documentation as possible with Mixxx. Tell me what's not working for you...

**Comment by HorstBaerbel** [comment]
Plus I would welcome improvements, but make sure you've got the latest version. Maybe it is also a good idea to pull-request on the reloop_mixage forks (don't forget the docs).

**Comment by gqzomer** [comment]
I used to the mapping from the reloop_mixage request you mentioned and it worked for me without any issues so no problems there. I am currently working on a few improvements and additions to the mapping. For now I made the following changes

- Using dicts instead of lists to store channel parameters, they can be retrieved using the group variable
- added the option to select and activate an effect for each slot on the effect rack using the controller
- added the option to open/close the library view using a double press of the trax button
- expanded the xml to include all available key presses the controller can send (useful for later additions)
- fixed an issue which caused the scratching to be active as long as the jog wheel was being turned even after the top was no longer being pressed
- added some additional led bindings

I also plan to eventually add support for decks 3 and 4 by using the midi channel switch on the back of the controller once I finalized the other changes and cleaned up the code. I believe this should be nothing more than a copy and paste of the xml mappings and change in the midi opcode for channel 3 and 4. If you are willing to collaborate on improving the mappings I'm up for it

**Comment by HorstBaerbel** [comment]
>     * Using dicts instead of lists to store channel parameters, they can be retrieved using the group variable
Sounds great.

>     * added the option to select and activate an effect for each slot on the effect rack using the controller
Sounds good too. I was too lazy to do that...

>     * added the option to open/close the library view using a double press of the trax button
There is already a config option to maximize the library when turning the Trax button. Not moving the Trax button or selecting a track into a deck minimizes the library. The delay is adjustable.
I kind of like it that way, but turned if off by default because the Mixxx maintainers asked me to do so. Does your change keep the original behaviour intact?

>     * expanded the xml to include all available key presses the controller can send (useful for later additions)
>     * fixed an issue which caused the scratching to be active as long as the jog wheel was being turned even after the top was no longer being pressed
>     * added some additional led bindings
Great!

> I also plan to eventually add support for decks 3 and 4 by using the midi channel switch on the back of the controller once I finalized the other changes and cleaned up the code. I believe this should be nothing more than a copy and paste of the xml mappings and change in the midi opcode for channel 3 and 4.
I'm still struggling with two decks :D but yeah. Why not.

> If you are willing to collaborate on improving the mappings I'm up for it
Sure! Make sure you don't break anything, the pull request checks are successful (JS checks etc.) and document your changes in the manual for everybody (Syntax checks there too).

**Comment by gqzomer** [comment]
Great I'll start working on merging the changes into a pull request. For now you can already test the changes I made by using the new mappings in the fork of this repo [here](https://github.com/gqzomer/MixxxReloopMixageMappings).

> There is already a config option to maximize the library when turning the Trax button. Not moving the Trax button or selecting a track into a deck minimizes the library. The delay is adjustable.
I kind of like it that way, but turned if off by default because the Mixxx maintainers asked me to do so. Does your change keep the original behaviour intact?

In my current mapping I removed the logic to automaximize the library since I added the double press feature. But there should be no issues implementing both features side by side. I hope to open a pull request with all the changes soon enough.

**Comment by HorstBaerbel** [comment]
No. Please fork [my fork of the Mixxx repo](https://github.com/HorstBaerbel/mixxx/tree/reloop_mixage) and [my fork of the manuals repo](https://github.com/HorstBaerbel/manual/tree/2.3). **This here repo is outdated. Don't use that code!** That's what I wrote earlier.
I've invited you to the repos i mentioned to collaborate.

**Comment by gqzomer** [comment]
I think I didn't communicate clearly that I intent to open a pull request on the forks you mentioned. I just linked to a fork of this repo which I have been using during development of the changes. I was orginally planning on making a new branch our your fork of the mixxx repo (in-fact I already created the `reloop_mixage_addition` branch) But doing the pull request from my own fork of your fork might be the cleaner option.

**Comment by HorstBaerbel** [comment]
The `reloop_mixage_additions` branch sounds good to me. I just wanted to make sure you don't waste your time tinkering with this repo :+1:

**Comment by gqzomer** [comment]
I have finalized all my changed and created a pull request on the `reloop_mixage` branch. I'm curious what you think of the changes I made. The pull request can be found  [here](https://github.com/HorstBaerbel/mixxx/pull/1)

**Comment by HorstBaerbel** [comment]
I got lots of stuff to do this week and a party coming up next weekend where I want to apply my limited DJ skills ;). I'll check out the pull request next week, if that's ok with you.

**Comment by gqzomer** [comment]
No worries, the changes in the pull request are quite substancial so take the time you need to review them
